File: components/messages/android/java/src/org/chromium/components/messages/MessageUtilsBridge.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.messages;

import org.chromium.base.annotations.CalledByNative;
import org.chromium.base.annotations.JNINamespace;

/**
 * Java counterpart to MessageUtilsBridge. Enables C++ feature code to know
 * if it is okay to show message ui.
 */
@JNINamespace("messages")
public class MessageUtilsBridge {
    @CalledByNative
    private static boolean isA11yEnabled() {
        return MessageUtils.isA11yEnabled();
    }
}

File: components/messages/android/message_utils_bridge.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/messages/android/message_utils_bridge.h"

#include <jni.h>

#include "base/android/jni_android.h"
#include "components/messages/android/jni_headers/MessageUtilsBridge_jni.h"

namespace messages {

// static
bool MessageUtilsBridge::IsA11yEnabled() {
  JNIEnv* env = base::android::AttachCurrentThread();
  return Java_MessageUtilsBridge_isA11yEnabled(env);
}

}  // namespace messages